exporter/opentelemetry-exporter-prometheus-remote-write/tests/test_prometheus_remote_write_exporter.py

```diff
@@ -0,0 +1,337 @@
+            PrometheusRemoteWriteMetricsExporter(
+                endpoint="/prom/test_endpoint", tls_config=tls_config
+            )
+
+
+class TestConversion(unittest.TestCase):
+    # Initializes test data that is reused across tests
+    def setUp(self):
+        self._test_metric = Counter(
+            "testname", "testdesc", "testunit", int, None
+        )
+        self._exporter = PrometheusRemoteWriteMetricsExporter(
+            endpoint="/prom/test_endpoint"
+        )
+
+        def generate_record(aggregator_type):
+            return ExportRecord(
+                self._test_metric, None, aggregator_type(), Resource({}),
+            )
+
+        self._generate_record = generate_record
+
+        def converter_method(record, name, value):
+            return (type(record.aggregator), name, value)
+
+        self._converter_mock = mock.MagicMock(return_value=converter_method)
+
+    # Ensures conversion to timeseries function works with valid aggregation types
+    def test_valid_convert_to_timeseries(self):
+        timeseries_mock_method = mock.Mock(return_value=["test_value"])
+        self._exporter.convert_from_sum = timeseries_mock_method
+        self._exporter.convert_from_min_max_sum_count = timeseries_mock_method
+        self._exporter.convert_from_histogram = timeseries_mock_method
+        self._exporter.convert_from_last_value = timeseries_mock_method
+        self._exporter.convert_from_value_observer = timeseries_mock_method
+        test_records = [
+            self._generate_record(SumAggregator),
+            self._generate_record(MinMaxSumCountAggregator),
+            self._generate_record(HistogramAggregator),
+            self._generate_record(LastValueAggregator),
+            self._generate_record(ValueObserverAggregator),
+        ]
+        data = self._exporter.convert_to_timeseries(test_records)
+        self.assertEqual(len(data), 5)
+        for timeseries in data:
+            self.assertEqual(timeseries, "test_value")
+
+        no_type_records = [self._generate_record(lambda: None)]
+        with self.assertRaises(ValueError):
+            self._exporter.convert_to_timeseries(no_type_records)
+
+    # Ensures conversion to timeseries fails for unsupported aggregation types
+    def test_invalid_convert_to_timeseries(self):
+        no_type_records = [self._generate_record(lambda: None)]
+        with self.assertRaises(ValueError):
+            self._exporter.convert_to_timeseries(no_type_records)
+
+    # Ensures sum aggregator is correctly converted to timeseries
+    def test_convert_from_sum(self):
+        sum_record = self._generate_record(SumAggregator)
+        sum_record.aggregator.update(3)
+        sum_record.aggregator.update(2)
+        sum_record.aggregator.take_checkpoint()
+
+        self._exporter.create_timeseries = self._converter_mock()
+        timeseries = self._exporter.convert_from_sum(sum_record)
+        self.assertEqual(timeseries[0], (SumAggregator, "testname", 5))
```

Comment:
Here you used `SumAggregator` in [this](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/207/files#diff-842c4d8d0cb6c63a88f2c4850fb0162d7f19e1d4b929af3c56a67d09c782f922R184) line. `3` and `2` from [these](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/207/files#diff-842c4d8d0cb6c63a88f2c4850fb0162d7f19e1d4b929af3c56a67d09c782f922R185-R186) lines add up to `5`. It is natural then for the test reader to find `SumAggregator` and `5` in the assertion. But where did `"testname"` was set in this test case? That is not that obvious and is a bit confusing. You may or may not want to refactor this test case to avoid that, just keep that in mind that whoever reads a test case expects to find initial conditions being set, operations on objects under test being done and finally assertions and the assertions should make sense based on the initial conditions set previously.

Reply:
Fixed by replacing generate_record method with ExportRecord constructor for clarity on contents of record.